server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/CassandraUploadUsageRepository.java

```diff
@@ -0,0 +1,68 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.cassandra.upload;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+
+import org.apache.james.core.Username;
+import org.apache.james.core.quota.QuotaComponent;
+import org.apache.james.core.quota.QuotaSizeUsage;
+import org.apache.james.core.quota.QuotaType;
+import org.apache.james.jmap.api.upload.UploadUsageRepository;
+import org.apache.james.mailbox.cassandra.quota.CassandraQuotaCurrentValueDao;
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraUploadUsageRepository implements UploadUsageRepository {
+
+    private CassandraQuotaCurrentValueDao cassandraQuotaCurrentValueDao;
+
+    @Inject
+    public CassandraUploadUsageRepository(CassandraQuotaCurrentValueDao cassandraQuotaCurrentValueDao) {
+        this.cassandraQuotaCurrentValueDao = cassandraQuotaCurrentValueDao;
+    }
+
+    @Override
+    public Publisher<Void> increaseSpace(Username username, QuotaSizeUsage usage) {
```

Comment:
Publisher -> Mono
more easier when somewhere invokes it

Reply:
I saw in your code you use Publisher as well. Could you explain why?
```
    public Publisher<Void> delete(UploadId id, Username user) {
        return uploadDAO.delete(user, id);
    }
```

Reply:
yes, it should be better if `public Mono<Void> delete`, my lazy when IDE auto-suggests complete code, 

My comment is because I see the `resetSpace` method use `Mono` to wrap the publisher in the same "internal" class

```
return Mono.from(getSpaceUsage(username)).flatMap(quotaSizeUsage -> Mono.from(decreaseSpace(username, quotaSizeUsage)))
            .then(Mono.from(increaseSpace(username, usage)));
```

it can be 
```
return getSpaceUsage(username).flatMap(quotaSizeUsage -> decreaseSpace(username, quotaSizeUsage))
            .then(increaseSpace(username, usage));
```

Reply:
Actually this is the thing I has been confused so far. If we know for sure the return type (Mono or Flux), we should use Mono or Flux instead of Publisher, right?

Reply:
Publisher -> API definition
Implementation -> Mono or Flux

Reply:
yes, 
- Mono, Flux for java implement, 
- SMono, SFlux for scala implement
- Publisher for interface/ trait

Reply:
Yes not always respected (but should try best to do so)
I think the rationale back in the day was well if somebody use something else than reactor, he could still do his own implem with his own reactive framework other than Reactor... 

Think of James as a big toolbox trying to be flexible enough to let people customize it a bit the way they want it :)